src/views/collection-view/components/collection-meta-progress/index.tsx

```diff
@@ -0,0 +1,98 @@
+import { useMemo } from 'react'
+import ButtonLink from 'components/button-link'
+import { IconCollections16 } from '@hashicorp/flight-icons/svg-react/collections-16'
+import { IconCheckCircleFill16 } from '@hashicorp/flight-icons/svg-react/check-circle-fill-16'
+import { useCollectionProgress } from 'hooks/progress'
+import { Collection } from 'lib/learn-client/types'
+import { parseCollectionProgress } from './helpers'
+import ProgressBar from 'components/progress-bar'
+import s from './collection-meta-progress.module.css'
+
+/**
+ * Displays collection progress status and CTA.
+ *
+ * Without authentication,
+ * we show a "Start" CTA which links to the first tutorial in the collection.
+ *
+ * When authenticated and one or more tutorials in the collection is "complete",
+ * we show a "Continue" CTA which links to the next not-"complete" tutorial,
+ * as well as a progress bar based on the count of tutorials done.
+ *
+ * When authenticated and all tutorials in the collection are "complete",
+ * we show a "Review" CTA which links to the first tutorial in the collection.
+ */
+function CollectionMetaProgress({ collection }: { collection: Collection }) {
```

Comment:
I've split this out as a separate component, it's used in `CollectionMeta`.

"What to render" logic is mainly handled in `parseCollectionProgress`, based on the `progressData` from `useCollectionProgress`.